### PR TITLE
Adding `discid`

### PR DESCRIPTION
**Pull request by HeavyTony2**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by HeavyTony2** [comment]
@conda-forge/help-python, The failed test in windows is because of the directory to find the DLL on which it depends.

**Comment by HeavyTony2** [comment]
Ping @conda-forge/help-python

**Comment by HeavyTony2** [comment]
Ping @conda-forge/help-python

**Comment by HeavyTony2** [comment]
Ping @conda-forge/help-python

**Comment by HeavyTony2** [comment]
https://github.com/orgs/conda-forge/teams/help-python, The failed test in windows is because of the directory to find the DLL on which it depends.

**Comment by ocefpaf** [comment]
The logs are gone. Closing and re-opening to toggle the CIs so we can see if the Windows problem persists.

**Comment by HeavyTony2** [comment]
> The Windows error is:
> 
> ```
> FileNotFoundError: Could not find module 'discid.dll' (or one of its dependencies). Try using the full path with constructor syntax.
> ```
> 
> I wonder if that is a problem in the `libdiscid` dependency. Do you want to skip Windows for now and investigate it later? If so, see https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-packages-with-os-specific-dependencies

I think the problem is about how to locate the libdiscId dep on Windows. Your suggestion seems perfect. How should we procede? I'm not yet familier with no-arch with os-specific-dependancies.

File: recipes/discid/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - python >=3.6
+    - pip
+  run:
+    - python >=3.6
+    - libdiscid
```
Comment (ocefpaf) [inline recipes/discid/meta.yaml:23]: This should make it noarch but installable only on Linux and macOS.

```suggestion
    - libdiscid
    - __unix  # [unix]
```

**Pull request merged**